libdevcore/vector_ref.h

```diff
@@ -76,10 +77,15 @@ class vector_ref
 	void cleanse()
 	{
 		static unsigned char s_cleanseCounter = 0;
+		size_t count = 0;
+		static std::mutex x_cleanseCounter;
```

Comment:
I think we can make it `std::atomic<unsigned char>` instead of using mutex

Reply:
I'll do this.